symfit/core/models.py

```diff
@@ -496,10 +499,20 @@ def __init__(self, model, independent_vars=None, params=None, **kwargs):
             self.connectivity_mapping = {var: set(independent_vars + params)
                                          for var in model}
         elif connectivity_mapping:
-            self.connectivity_mapping = connectivity_mapping
             if not isinstance(model, Mapping):
                 raise TypeError('Please provide the model as a mapping, '
                                 'corresponding to `connectivity_mapping`.')
+            elif connectivity_mapping.keys() != model.keys():
+                # Infer the connectivity mapping corresponding to the symbolical
+                # part automatically
+                sub_model = {}
+                for var, expr in model.items():
+                    if isinstance(expr, sympy.Basic):
+                        sub_model[var] = expr
+                sub_model = BaseModel(sub_model)
+                connectivity_mapping.update(sub_model.connectivity_mapping)
```

Comment:
I have a feeling this update is not good enough when there are interdependent components involved. Can't come up with an example at the moment though...

Reply:
I cannot guarantee that it is not possible, but I have a feeling it would be very hard and luckily the error messages will be very helpful in this case: something like you forgot to give a value to this keyword argument. So it should be easily debug-able. I don't think it can lead to fundamental conflicts though, since the user always has the final say with the update happening last.

Reply:
The comment is wrong. User provided connectivity_mapping gets overwritten by sub_model connectivity_mapping. I do like the case where user input is prioritized over auto-generated data.